### PR TITLE
feat(skills): add story-lifecycle skill (#2308)

### PR DESCRIPTION
**Pull request by kritikagarg**
> **Note / PR Context:** My original PR (#2726) that introduced both `dev-team` and `story-lifecycle` was accidentally closed. Since `dev-team` was merged separately via #2309, this PR delivers the remaining `story-lifecycle` skill.

## What Changed

Added the **`story-lifecycle`** PM/delivery skill implementing the remaining requirement from issue #2308, inspired by the BMAD method:

- **`story-lifecycle`** — file-based epic → story → sprint delivery loop tracked in `.delivery/` markdown files. No external PM tool required; works with any AI harness. Defines a full story status schema (`backlog | ready | in-progress | review | done | closed | blocked`) with `closed` as the explicit terminal sprint-archived state. Agents load `CLAUDE.md` before implementing each story.

**Omitted skills rationale:**
- **`dev-team`** intentionally omitted — already implemented and merged via #2309.
- **`project-context`** intentionally omitted — `CLAUDE.md` already serves as ECC's shared root-context convention, and PR #2309 folded `PROJECT-CONTEXT.md` generation directly into `dev-team`.

**Also includes:**
- Content assertions and structure tests in `tests/skills/story-lifecycle.test.js`.
- Catalog sync updating overall skill count to **285**.

## Why This Change

Issue #2308 requested these skills to fill a gap in ECC's project management coverage. ECC had strong technical workflow skills (`tdd-workflow`, `council`, `architecture-decision-records`) but no cohesive delivery planning layer. 

The two skills work together: `dev-team` (#2309) surfaces multi-role alignment on a problem, and `story-lifecycle` (this PR) translates that alignment into sprint-ready deliverables.

## Testing Done

- [x] Manual testing completed
- [x] Automated tests pass locally (`node tests/run-all.js`)
- [x] Edge cases considered and tested

All skills pass `node scripts/ci/validate-skills.js --strict` (286 skill directories validated).
Content tests verify frontmatter, section structure, cross-references, output format, anti-patterns, sprint-close synchronization, and examples.

## Type of Change

- [x] `feat:` New feature

## Security & Quality Checklist

- [x] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [x] JSON files validate cleanly
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## If you added a skill, command, agent, hook, or CLI tool

- [x] Regenerated the catalog (`npm run catalog:sync`) — skill count updated in `README.md`, `README.zh-CN.md`, `AGENTS.md`, `docs/zh-CN/`, `.claude-plugin/plugin.json`, `.claude-plugin/marketplace.json`
- [ ] Registered in `package.json` (`files`), `manifests/install-components.json`, `manifests/install-modules.json`, and `agent.yaml` — skills are markdown-only with no scripts; maintainers to confirm if manifest entries are required
- [ ] Cross-harness surfaces updated (Codex `.agents/skills/`) — out of scope for this PR; can be added by maintainers if needed

## Documentation

- [x] README updated (skill count via catalog sync)

Closes #230

### CODE REVIEW

File: tests/skills/story-lifecycle.test.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+console.log('\n=== story-lifecycle skill tests ===\n');
+
+try {
+  skill = fs.readFileSync(skillPath, 'utf8');
+  frontmatterMatch = skill.match(/^---\n([\s\S]*?)\n---/);
+} catch (err) {
+  console.log(`  ✗ fixture: SKILL.md must be readable`);
+  console.log(`    Error: ${err.message}`);
+  console.log(`\nResults: Passed: 0, Failed: 1`);
```
[the remark this answers is not in the document]
Comment (kritikagarg) [reply tests/skills/story-lifecycle.test.js:36]: This finding is not valid — skipping.
Reason: the combined Results: Passed: N, Failed: N format is the dominant convention across the suite (114 test files use it; 0 use the separate Passed:/Failed: line format). The finding's line references are also stale — line 285 is now a comment inside a test (the summary output is at line 333), and both actual summary sites (line 36 and 333) already use the combined format. tests/run-all.js:100-101 parses via /Passed:\s*(\d+)/ and /Failed:\s*(\d+)/, which match both formats, so exit behavior is already preserved. Changing to the non-conventional format would make this file inconsistent with 114 sibling tests. No change made; validators already pass.

File: AGENTS.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Everything Claude Code (ECC) — Agent Instructions
 
-This is a **production-ready AI coding plugin** providing 68 specialized agents, 284 skills, 94 commands, and automated hook workflows for software development.
+This is a **production-ready AI coding plugin** providing 68 specialized agents, 285 skills, 94 commands, and automated hook workflows for software development.
```
[the remark this answers is not in the document]
Comment (kritikagarg) [reply AGENTS.md:3]: 285. The directory scan (skills/*/SKILL.md) and the validate-skills.js validator both report 285. The "286" figure was the incorrect catalog number we fixed earlier via catalog.js --write, which synced docs/plugin manifests down to the real 285.

**Comment by haelyra** [comment]
Thank you for taking on a repository-native story lifecycle. The epic, story, sprint, recovery, and path-safety work is valuable, and it reinforces the community demand for lightweight local delivery planning.

We are closing this version because PR #2311 already carries the same story-lifecycle contribution, has completed the maintainer hardening and approval path, and preserves the focused implementation without this branch’s unrelated dependency and release-surface drift. We appreciate the work here and would be glad to see future focused contributions in this planning area.
